### PR TITLE
rm task from storage when terminated

### PR DESCRIPTION
**Pull request by meganrogge**
The wrong key was being used, so the task wouldn't get deleted when it was terminated. https://github.com/microsoft/vscode/pull/197381/files#diff-f77e08d48fa5a43f6cd385c0bac2c04ee2123565ead5f0f4b60e23c8af821b76R344

Also, aptly renamed `getRecentlyUsedKey` -> `getKey` and added a bunch of logs behind a setting

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 1]
```diff
@@ -2329,14 +2344,20 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
...
+				this._log(nls.localize({ key: 'TaskSystem.invalidTaskJsonOther', comment: ['Message notifies of an error in one of several places there is tasks related json, not necessarily in a file named tasks.json'] }, 'Error: The content of the tasks json in {0} has syntax errors. Please correct them before executing a task.', location));
 				this._showOutput();
 				return { config, hasParseErrors: true };
 			}
 		}
 		return { config, hasParseErrors: false };
 	}
 
+	private _log(value: string, verbose?: boolean): void {
+		if (verbose && this._configurationService.getValue(TaskSettingId.VerboseLogging) || !verbose) {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:2356]: Little simpler:

```suggestion
		if (!verbose || this._configurationService.getValue(TaskSettingId.VerboseLogging)) {
```

File: src/vs/workbench/contrib/tasks/browser/task.contribution.ts [hunk 2]
```diff
@@ -553,5 +553,10 @@ configurationRegistry.registerConfiguration({
 			],
 			default: 'always',
 		},
+		[TaskSettingId.VerboseLogging]: {
+			type: 'boolean',
+			description: nls.localize('task.verboseLogging', "Enable verbose logging for tasks."),
+			default: false
+		},
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/task.contribution.ts:560]: Can we use trace logging instead of a new setting?

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/contrib/tasks/browser/task.contribution.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/browser/task.contribution.ts:560]: no, I investigated that a few iterations ago and ultimately gave up.

Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/browser/task.contribution.ts:560]: https://github.com/microsoft/vscode/pull/192195

**Review by bpasero — APPROVED** [review]

**Pull request merged**